vscode-trace-extension/src/utils/backend-tsp-client-provider.ts

```diff
@@ -67,9 +69,16 @@ export async function updateNoExperimentsContext(): Promise<void> {
     return;
 }
 
-async function getExternalUriFromUserSettings(): Promise<vscode.Uri> {
+async function getExternalUriFromUserSettings(isFrontEnd: boolean): Promise<vscode.Uri> {
```

Comment:
Question: have you considered using an Enum instead of a Boolean? It might make the code easier to understand, specially from the calling code. 

Same could apply for getTraceServerUrl() and  getTspClientUrl()

Reply:
I tried applying a enum. Let me know if you like it better now.

Reply:
Done. The branch is also squashed, so if there are no other concerns/comments it should be ready to go